src/Cmts.ml

```diff
@@ -256,9 +256,11 @@ let is_adjacent t (l1 : Location.t) (l2 : Location.t) =
           && Position.column l1.loc_start <= Position.column l2.loc_start
       | _ -> false )
 
-(** Heuristic to choose between placing a comment after the previous loc or
-    before the next loc. Places comment after prev loc only if they are
-    "adjacent", and the comment and next loc are not "adjacent". *)
+(** Heuristic to choose between placing a comment after the previous
+    location or before the next one. Places comment after the previous
+    location only if they are "adjacent", and the comment and the next
+    location are not "adjacent" or the comment and the previous location are
+    on the same line (different from the next location) *)
```

Comment:
@jberdine you said on the parent PR that the documentation needed to be updated, I personally find this comment too long and confusing, are you sure there is a smart way to document this without paraphrasing the code? (can we just remove it?)

Reply:
There is the point that how can people expect to understand what the formatter does if we can't describe the heuristic used. So I think that there is definite value in the description, though probably more in user-facing documentation than in code documentation.

That said, if the best explanation is unusable for user-facing documentation and amounts to paraphrasing the code, then we might as well remove it (except the first sentence).